### PR TITLE
Refactor and cleanup of webkit_inspection_protocol.dart

### PR DESCRIPTION
**Pull request by DrMarcII**
Move each domain into a separate part.
Delete some unused/unfinished code.
Simplify code by removing _Event and WipObject classes.
Change some code to use async/await.
Add generic public onNotification stream and sendCommand method to WipConnection to support planned debug port multiplexor.

### CODE REVIEW

**Comment by DrMarcII** [comment]
I did my best not to change the API exposed by this, but there may be some corner cases where it has changed.

File: pubspec.yaml [hunk 1]
```diff
@@ -6,4 +6,6 @@ homepage: https://github.com/google/webkit_inspection_protocol.dart
 author: Devon Carew <devoncarew@google.com>
 
 environment:
-  sdk: '>=1.0.0 <2.0.0'
+  sdk: '>=1.9.0 <2.0.0'
+dependencies:
+  logging: '0.11.1'
```
Comment (devoncarew) [inline pubspec.yaml]: ==> '^0.11.1' (or '^0.11.0')

**Comment by devoncarew** [comment]
lgtm; can you update from master?

File: pubspec.yaml (hunk 1 above)
Comment (DrMarcII) [inline pubspec.yaml]: done.

**Comment by DrMarcII** [comment]
I am going to create a new PR with cleaner history.

**Pull request closed without merging**